README.md
# landing-page-components
> basic, styled landing page components for the busy developer

# Install
```
$ npm i --save landing-page-components
```

# Usage
Landing Page components is a collection of components that can be used collectively to build a landing page.

To include any one of the components in your project, import into a React App with
```
import { componentName } from 'landing-page-components';
```
and you're on your way.

# Reference
Please use the src/index.js in the demo folder of the Demo App as a reference of where the Landing Page components can be used.

# Running the Demo App
```
$ npm start
```

Then navigate to http://localhost:3000/ to see the components being used.


changelog.md
# Changelog

## [1.0.8] - 2018-08-28
### Added
- Adds a background-size to the SplitPicDiv component.
- Defaults the background-size to 'cover'

## [1.0.7] - 2018-08-26
### Added
- adds id to be able to be passed in to the StyledIcon components

## [1.0.6] - 2018-08-25
### Mistype
- Mistyped package information and re-released with a different number (1.0.6)

## [1.0.5] - 2018-08-25
### Fixes
- An issue where not both StyledNav types were having their href property passed in correct

## [1.0.4] - 2018-08-23
### Added
- Unique keys to sections inside of the Section and Navbar js files when mapping over icons and links to print html.

## [1.0.3] - 2018-08-20
### Added
- enables icons to be added to the Icon div
- Adds css emoji support for the Icon Divs so that any font icons that are css content unicode strings can be used

## [1.0.2] - 2018-08-13
### Added
- Reasonable Spacing around the StyledHeader
- Changes the StyledHeader component to be able to wrap it around child elements

## [1.0.1] - 2018-08-09
### Added
- fixes error where you couldn't change the background position of the StyledHeader

## [1.0.0] - 2018-08-05
### Added
- Basic landing page components
- A finished demo page consuming the landing-page-components


src/Navbar.js
import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
	width: 100%;
	max-width: 900px;
	display: flex;
	align-items: center;
	justify-content: space-between;
	box-sizing: border-box;
	text-align: left;
	padding: 0px 40px;
	height: 100%;
`;

const StyledNavButton = styled.a`
	&:-moz-focus-inner { border: 0; padding: 0; margin:0; }
	background-color: rgb(0, 123, 255);
	border-radius: 8px;
	border: 0px solid rgb(0, 123, 255);
	padding: 0px 8px;
	cursor: pointer;
	color: white;
	height: 40px;
	font-size: 20px;
	&:hover {
		background-color: #0069d9;
		border-color: #0062cc;
		transition: background-color .15s ease-in-out;
	}
	display: flex;
	align-items: center;
`;

const StyledNavLink = styled.a`
	font-size: 20px;
	height: 40px;
	display: flex;
	align-items: center;
`;
	
const Navbar = ({className, links}) => {
	return (
		<nav className={className}>
			<Container>
				{links.map((link, index) => {
					let {href, text, isButton} = link;
					return (isButton ? <StyledNavButton href={href} key={index}>{text}</StyledNavButton> : <StyledNavLink href={href} key={index}>{text}</StyledNavLink>);
				})}
			</Container>
		</nav>
	);
};

export const StyledNav = styled(Navbar)`
	display: flex;
	box-sizing: border-box;
	align-items: center;
	justify-content: center;
	flex-wrap: wrap;
	height: 60px;
`;


src/Overlay.js
import React from 'react';
import styled, { css } from 'styled-components';

const TextDiv = styled.div`
	height: 205px;
`;

const Header = ({className, children}) => {
	return (
		<header>
			<div className={className}>
				{children}
			</div>
		</header>
	);
};

export const StyledHeader = styled(Header)`
	box-sizing: border-box;
	font-size: 16px;
	color: rgba(0, 0, 0, 0);
	height: 223px;

	${props => props.imageUrl && css`
		background-image: url(${props.imageUrl});
		background-origin: padding-box;
		background-size: cover;
		background-position-x: ${props => props.backgroundPositionX ? props.backgroundPositionX : ''};
		background-position-y: ${props => props.backgroundPositionY ? props.backgroundPositionY : ''};
		padding: 250px 0px;
		text-align: center;
		position: relative;
	`};
`;


demo/src/index.js
import React, {Component} from 'react'
import {render} from 'react-dom'
import sizes from 'point-breaks';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons'
import { StyledNav, StyledHeader, StyledIconSection, SplitSection, SplitPicDiv, SplitTextDiv, SplitTextTitle, SplitTextBody,
	CenteredSection, CenteredTitle,
	StyledEmailSection,
	StyledEmailInput, StyledSubmitButton,
	FooterSection, FooterLinksDiv,
	StyledListOfLinks,
	CopyrightText, StyledListLink
} from '../../src';
import './App.css';

library.add(fab);

class App extends Component {
  constructor(props) {
    super(props);
	this.state = {width: 0};
  }

  updateWidth() {
	this.setState({ width: window.innerWidth });
  }

  componentDidMount() {
	this.updateWidth();
	window.addEventListener("resize", this.updateWidth.bind(this));
  }

  componentWillUnmount() {
	window.removeEventListener("resize", this.updateWidth.bind(this));
  }

  render() {
	const smallTabletWidth = Number(sizes.smallTablet.width.slice(0, -2));
	const secondDivText = "Lorem ipsum dolor amet adaptogen thundercats keffiyeh pop-up lomo ramps. Butcher hella marfa, narwhal yr microdosing ramps."
	const determineSecondSplitDiv = (orderInput) => {
		return (
			<SplitTextDiv order={orderInput}>
				<SplitTextTitle>
					Totally Furry Kitties
				</SplitTextTitle>
				<SplitTextBody>{secondDivText}</SplitTextBody>	
			</SplitTextDiv>
		);
	};

	const iconArr = [{
		title: 'Icon1',
		pic: "\u{1f40e}",
		desc: "This is a shady icon 1"
	},
	{
		title: 'Icon2',
		pic: "\u{1f40e}",
		desc: "This is a shady icon 2"
	},
	{
		title: 'Icon3',
		pic: "\u{1f40e}",
		desc: "This is a shady icon 3"
	}];

    return (
      <div className="App">
		<StyledNav links={[{href: "google.com", text: "Home"}, {href: "amazon.com", text: 'Sign Up', isButton: true}]}></StyledNav>
		<StyledHeader imageUrl={"https://placekitten.com/1200/945"} backgroundPositionY="50%">
			Discover Kittens
		</StyledHeader>
        
		<StyledIconSection icons={iconArr} id="#about" />
		<SplitSection>
			<SplitTextDiv order={'left'} >
				<SplitTextTitle>
					Totally Kitties
				</SplitTextTitle>
				<SplitTextBody>
					Pee in the shoe lick the curtain just to be annoying but the dog smells bad show belly.
				</SplitTextBody>
			</SplitTextDiv>
			<SplitPicDiv order={'right'} imageUrl={'https://placekitten.com/1200/480'} />
		</SplitSection>
		<SplitSection>
			{ this.state.width < smallTabletWidth? determineSecondSplitDiv('left') : determineSecondSplitDiv('right')}
			{ this.state.width < smallTabletWidth ? <SplitPicDiv order={'right'} imageUrl={'https://placekitten.com/1200/480'} /> : <SplitPicDiv order={'left'} imageUrl={'https://placekitten.com/1200/480'} />}
		</SplitSection>
		<SplitSection>
			<SplitTextDiv order={'left'} >
				<SplitTextTitle>
					Totally Cats
				</SplitTextTitle>
				<SplitTextBody>
					These cats are the best cats I have ever seen!  Never before have cats been so interesting to me and taught me sooooo much!
				</SplitTextBody>
			</SplitTextDiv>
			<SplitPicDiv order={'right'} imageUrl={'https://placekitten.com/1200/480'} />
		</SplitSection>
		<CenteredSection>
			<CenteredTitle>
				Call toAction Section!!	
			</CenteredTitle>
			<StyledEmailSection>
				<StyledEmailInput />
				<StyledSubmitButton>Sign Up!</StyledSubmitButton>
			</StyledEmailSection>
		</CenteredSection>
		<FooterSection>
			<FooterLinksDiv>
				<StyledListOfLinks style={{'display': 'flex'}}>
					<StyledListLink first={true} linkUrl="https://www.google.com" linkText="Google" />
					<StyledListLink linkUrl="https://www.xkcd.com" linkText="Xkcd" />
					<StyledListLink last={true} linkUrl="https://www.youtube.com" linkText="Youtube" />
				</StyledListOfLinks>		
				<CopyrightText>© Ru-Lai Development 2018. All Rights Reserved.</CopyrightText>
			</FooterLinksDiv>			
		</FooterSection>
      </div>
    );
  }
}

class Demo extends Component {
  render() {
    return <div>
      <App />
    </div>
  }
}

render(<App/>, document.querySelector('#demo'))


src/Section.js
import React from 'react';
import styled from 'styled-components';
import sizes from 'point-breaks';

const IconHolder = styled.div`
	display: flex;
	justify-content: space-around;
	margin: 0 auto;
	flex-wrap: wrap;
	max-width: 1140px;
`;

const IconTitle = styled.h3`
	font-size: 28px;
	text-align: center;
	font-weight: 700;
	line-height: 12px;
`;

const IconDesc = styled.div`
	font-size: 20px;
	font-weight: 300;
	line-height: 30px;
	text-align: center;
`;

const IconDiv = styled.div`
	max-width: 320px;
	padding: 20px 0;
	box-sizing: border-box;
`;

const IconPicDiv = styled.div`
	text-align: center;
	:before {
		content: '${props => props && props.picContent ? props.picContent : ''}';
		font-size: 4.5rem;
	}
`;

const IconSection = ({className, icons, id}) => {
	return (
		<section className={className} id={id}>
			<IconHolder>
				{
					icons.map((icon, index) => {
						return (
							<IconDiv key={index}>
								<IconPicDiv picContent={icon.pic} />
								<IconTitle>{icon.title}</IconTitle>
								<IconDesc>{icon.desc}</IconDesc>
							</IconDiv>
						);
					})
				}
			</IconHolder>
		</section>
	);
}

export const StyledIconSection = styled(IconSection)`
	padding: 7rem 0;
	box-sizing: border-box;
`;

export const SplitSection = styled.section`
	display: flex;
	flex-direction: row;
	@media (max-width: ${sizes.smallTablet.width}) {
		flex-direction: column;
		justify-content: center;
	}
`;

export const SplitPicDiv = styled.div`
	box-sizing: border-box;
	flex-basis: 50%;
	max-width: 50%;
	order: ${props => props && props.order === 'right' ? 2 : 1};
	min-height: 480px;
	background-image: url(${props => props && props.imageUrl ? props.imageUrl : ''});
	background-repeat: no-repeat;
	background-size: ${props => props && props.backgroundSize ? props.backgroundSize : 'cover'}
	@media (max-width: ${sizes.smallTablet.width}) {
		width: 100%;
		max-width: 100%;
	}
`;

export const SplitTextDiv = styled.div`
	flex-basis: 50%;
	padding: 112px;
	display: flex;
	flex-direction: column;
	line-height: 24px;
	max-width: 50%;
	max-height: 480px;
	order: ${props => props && props.order === 'right' ? 2 : 1};
	width: 538px;
	justify-content: center;
	text-align: left;
	@media (max-width: ${sizes.smallTablet.width}) {
		width: 100%;
		max-width: 100%;
	}
	box-sizing: border-box;
`;

export const SplitTextTitle = styled.h2`
	box-sizing: border-box;
	font-size: 32px;
	line-height: 38.2px;
	margin-bottom: 8px;
	margin-top: 0px;
`;

export const SplitTextBody = styled.p`
	font-size: 20px;
	font-weight: 300;
`;

export const CenteredSection = styled.section`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	min-height: 340px;
	background-color: ${props => props && props.backgroundColor ? props.backgroundColor : '#0e1a2a'}
	color: ${props => props && props.color ? props.color : 'whitesmoke'}
`;

export const CenteredTitle = styled.h2`
	font-size: 32px;
	line-height: 38.2px;
	margin-bottom: 8px;
	margin-top: 0px;
	padding-bottom: 0.5rem;
	box-sizing: border-box;
`;

export const CenteredTextBody = styled.p`
	width: 516px;
	font-size: 20px;
	font-weight: 300;
	text-align: center;
`;

export const StyledEmailSection = styled.div`
	display: flex;
	height: 3rem;
`;

export const StyledEmailInput = styled.input.attrs({
	type: 'email',
	placeholder: 'Please enter your email'
})`
	margin-right: 10px;
	width: 300px;
	color: black;
	border-radius: .3rem;
	padding: .3rem 1rem;
	box-sizing: border-box;
`;

export const StyledSubmitButton = styled.button`
	color: white;
	padding: .3rem;
	border: 1px solid transparent;
	border-radius: .3rem;
	background-color: #007bff;
	padding: .6rem;
	&:hover {
		background-color: #0069d9;
		transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
	}
	box-sizing: border-box;
`;

export const FooterSection = styled.section`
	display: flex;
	justify-content: flex-start;
	align-items: center;
	min-height: 150px;
	background-color: ${props => props.backgroundColor ? props.backgroundColor : 'whitesmoke'}
	color: ${props => props.color ? props.color : 'black'}
	padding-left: 3rem;
	@media (max-width: ${sizes.smallTablet.width}) {
		justify-content: center;
		padding-left: 0;
	}
	box-sizing: border-box;
`;

export const FooterLinksDiv = styled.div`
	
`;

export const MediaLinksDiv = styled.div`
	grid-row-start: 3;
	grid-column-start: 4;
	padding: 20px 0px;
	display: flex;
	box-sizing: border-box;
`;

const Link = ({linkUrl, className, children}) => {
	return (
		<a href={linkUrl} className={className}>{children}</a>
	)
}

export const MediaLink = styled(Link)`
	padding: 0 10px;
	box-sizing: border-box;
`;

export const CopyrightText = styled.p`
	color: #6c757d !important
	font-size: 12.8px;
	font-weight: 400;
`;

export const StyledListOfLinks = styled.ul`
	list-style: none;
	padding: 0px;
	display: 'flex';
	box-sizing: border-box;
`;

export const ListLink = ({linkUrl, linkText, className}) => {
	return (
		<li className={className}>
			<a href={linkUrl}>{linkText}</a>
		</li>
	);
};

export const StyledListLink = styled(ListLink)`
	font-size: 16px
	padding-right: ${props => props && props.last ? '0px' : '20px'};
	padding-left: ${props => props && props.first ? '0px' : '20px'};
	border-right: ${props => props && props.last ? '' : '1px solid black'};
	color: rgb(0, 123, 255);
	box-sizing: border-box;
`;
